src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.PlatformServices/Options/IAppOptionsProvider.cs

```diff
@@ -0,0 +1,27 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Altinn.App.Common.Models;
+
+namespace Altinn.App.PlatformServices.Options
+{
+    /// <summary>
+    /// Interface for providing <see cref="AppOptions"/>
+    /// </summary>
+    public interface IAppOptionsProvider
+    {
+        /// <summary>
+        /// The id/name of the options this provider supports ie. land, fylker, kommuner.
+        /// You can have as many providers as you like, but you should have only one per
+        /// id. 
+        /// </summary>
+        string Id { get; }
+
+        /// <summary>
+        /// Gets the <see cref="AppOptions"/> based on the provided options id and key value pairs.
+        /// </summary>
+        /// <param name="keyValuePairs">Key/value pairs to control what options to get.
+        /// When called from the options controller this will be the querystring key/value pairs.</param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        Task<AppOptions> GetAppOptionsAsync(Dictionary<string, string> keyValuePairs);
```

Comment:
Yes and yes. We had a quick discussion here now and the proposal is the following:
1. If a language query parameter is provided, we will pick up that in the controller and pass it as a separate parameter to the GetOptionsAsync method.
2. If no query param is provided, we'll fall back to the cookie provided and pick up the language from there.
3. We will change the frontend to pass inn the language as a query param.

Any thoughts?

Reply:
The current implementation will actually pass on any query parameters provided in the keyValuePairs parameter, but we wanted it to be explicit.

Reply:
When fetching from code lists from API or other sources, it is usually because we don't want to hard code the options into the app, and that probably means we don't want to include the translations in the `resources_nn.json` file either.

Reply:
Sorry for the extra comment, but I didn't see your reply before posting.

Yes! That definitely is what I'm thinking. An explicit parameter and a standard way across apps for this functionality.

Reply:
NP:) I'm on my way of adding it now. The only thing that bothers me a little bit is that the interface specifies language, which makes sense, but by that we sort of "promise" that we handle language - which we don't in our default implementation. So default implementation will deal with language as it used to, ie. leave it up to the frontend to request the language resources. At least for now.

I'm thinking we should allow for language support in the option files as well, something along the lines of: 
* countries_en.json
* countries_nb.json
That way we could easily create a Nuget package containing the code and resources needed to plug in support for country options just by registering the CountryOptionsProvider in startup. No need to manipulate language resource files. A lot of standard options could the be created allowing for easy re-use accross apps.

Reply:
Yea, I'm more looking at a nuget package that fetches options from a central store, so that updates don't require a redeploy of all apps. But the functionality required is the same, so I don't think there should be any difference.